ui_testing/testcases/basic_tests/test03_orders.py

```diff
@@ -750,3 +750,65 @@ def test017_update_shipment_date(self, save):
                 ' + Assert {} (current_shipment_date) == {} (order_shipment-date)'.format(current_shipment_date,
                                                                                           order_shipment_date))
             self.assertEqual(current_shipment_date, order_shipment_date)
+
+    @parameterized.expand(['save_btn', 'cancel'])
+    def test020_update_material_type_with_test_unit(self, save):
+        """
+
+        New: Orders: Material type Approach: I can update the material type
+        filed with test units records successfully
+
+        LIMS-3844
+        LIMS-3844
+        :return:
+        """
+
+        self.order_page.get_orders_page()
+        self.order_page.click_create_order_button()
+        order_url = self.base_selenium.get_url()
+        self.base_selenium.LOGGER.info(' + order_url : {}'.format(order_url))
+        self.order_page.set_new_order()
+        self.order_page.set_contact()
+        self.order_page.set_material_type(
+            material_type='r')
+        self.order_page.set_article(
+            article='a')
+        self.order_page.set_test_unit(
+            test_unit='f')
+        self.order_page.save(save_btn='order:save_btn')
+        self.order_page.get_orders_page()
+        orders_result = self.orders_page.result_table()
+        self.order_page.get_random_x(row=orders_result[0])
+        order_url = self.base_selenium.get_url()
+        self.base_selenium.LOGGER.info(' + order_url : {}'.format(order_url))
+        order_material_type = self.order_page.get_material_type()
+        self.order_page.set_material_type(
+            material_type='s')
+        new_material_type = self.order_page.get_material_type()
+        self.order_page.confirm_popup(force=True)
+        self.order_page.set_article(
+            article='a')
+        self.order_page.set_test_unit(
```

Comment:
what if there is no test uint for this article at all?

Reply:
@0xIslamTaha nothing occur because the test unit depend on the material type not on the article not such as test plan

Reply:
okay

Reply:
@dianaeldemerdash what if there is no article for this material type at all??

Reply:
@0xIslamTaha I didn't select the materiel type when I update it will still ( r ) same as the last choice so in this case I'm sure that this material type has article or what about the solution in your point of view.

Reply:
@dianaeldemerdash if u didnt wanna change the material type, then you have to option
1- Don't do set_material at all
2- set_material(self.get_material())